src/lib/Microsoft.Health.Fhir.R4.Ingest/Service/R4FhirImportService.cs

```diff
@@ -98,6 +98,23 @@ public virtual async Task<string> SaveObservationAsync(ILookupTemplate<IFhirTemp
             return result.Id;
         }
 
+        public override Task<string> CheckHealth()
+        {
+            SearchParams q = new SearchParams()
+                .ForSubject<Model.Patient>("IoMTHealthCheck")

```

Comment:
Would be better to do a get against a foundational resource that won't have any PII like StructureDefinition or SearchParameter.  I understand the desire is to have no search result hence an id that shouldn't exist but a call against a non-PII resource would be preferred.  A get or search against a structural component will have less overhead on the underlying service too.  This will still execute a complete search against the patients.

Reply:
There is a /health/check endpoint you can hit for the FHIR server as well but it doesn't require authentication.  We could consider adding a special one that behaves the same way but requires auth.  This wouldn't be in the open source though since it isn't a FHIR standard endpoint, rather an implementation detail on our end.

Reply:
That makes sense, I can change it to something without PII, for now.

Reply:
It would be fairly easy to add a controller to the FhirApplication that could do an "authentication health check". Might not be a bad idea so we can catch issues early on if they occur.

Reply:
I think the decision we need to make here is if the health check should be something that works against any FHIR server or if we want something specific when connecting to the Azure API for FHIR.